### PR TITLE
Add pyphysim

### PR DESCRIPTION
**Pull request by oblute**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by oblute** [comment]
@rluria14 agree to maintain?

File: recipes/pyphysim/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    - "*/libc.so.6"  # [linux]
+
+requirements:
+  host:
+    - pip
+    - python >=3.7,<4.0
+    - rtd-poetry
+  run:
+    - configobj
+    - matplotlib
```
Comment (xhochy) [inline recipes/pyphysim/meta.yaml:27]: If you don't need the Qt backend, use `matplotlib-base`.
```suggestion
    - matplotlib-base
```

File: recipes/pyphysim/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 0df90c4d8031503d69d2cbd13717daf34cab8295a73aaecdb70e08441cdeaaae
+
+build:
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+  missing_dso_whitelist:
+    - "*/libpthread.so.0"  # [linux]
+    - "*/libc.so.6"  # [linux]
```
Comment (xhochy) [inline recipes/pyphysim/meta.yaml:18]: This sounds like the package contains compiled code, that should be rather compiled here and probably makes it non-noarch.

**Comment by oblute** [comment]
@xhochy added the compiler which solved all the issues. ready to go when you are

**Comment by xhochy** [comment]
There is still a failure on Windows (I have no idea about that).

**Comment by oblute** [comment]
@xhochy yep, it's not working on windows due to environmental dependency issues. we've been told before to keep it as `noarch` when windows doesn't work due to a dependency.

**Comment by xhochy** [comment]
Recipes that require a compiler cannot be `noarch` as they contain arch-specific code. You need to skip Windows then,.

**Comment by oblute** [comment]
@xhochy @conda-forge/staged-recipes the selector for python isn't working when i switch. am i doing something wrong?

File: recipes/pyphysim/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 0df90c4d8031503d69d2cbd13717daf34cab8295a73aaecdb70e08441cdeaaae
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+  skip: True  # [win]
```
Comment (xhochy) [inline recipes/pyphysim/meta.yaml:15]:
```suggestion
  skip: True  # [win or py<37]
```

File: recipes/pyphysim/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - pip
+    - python >=3.7,<4.0
```
Comment (xhochy) [inline recipes/pyphysim/meta.yaml:22]:
```suggestion
    - python
```

File: recipes/pyphysim/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    - rtd-poetry
+  run:
+    - configobj
+    - matplotlib-base
+    - numba
+    - numpy
+    - python >=3.7,<4.0
```
Comment (xhochy) [inline recipes/pyphysim/meta.yaml:29]:
```suggestion
    - python
```

**Review by xhochy — COMMENTED** [review]
That's what the linter (and later conda-smithy) is looking for.

**Comment by oblute** [comment]
@xhochy @conda-forge/staged-recipes as seen above, i tried twice with selectors to replicate the pins, and it fails with the selectors. what am i doing wrong? i follow the above suggestion and still no success.

File: recipes/pyphysim/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 0df90c4d8031503d69d2cbd13717daf34cab8295a73aaecdb70e08441cdeaaae
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+  skip: True  # [win or py<37]
```
Comment (xhochy) [inline recipes/pyphysim/meta.yaml:15]: Shouldn't this be what you want? The error in CI is about a line that is only working for Python 3.8+
```suggestion
  skip: True  # [win or py<38]
```

**Pull request merged**